src/phocount.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2014, Brookhaven Science Associates, Brookhaven        
+ * National Laboratory. All rights reserved.                            
+ *                                                                      
+ * Redistribution and use in source and binary forms, with or without   
+ * modification, are permitted provided that the following conditions   
+ * are met:                                                             
+ *                                                                      
+ * * Redistributions of source code must retain the above copyright     
+ *   notice, this list of conditions and the following disclaimer.      
+ *                                                                      
+ * * Redistributions in binary form must reproduce the above copyright  
+ *   notice this list of conditions and the following disclaimer in     
+ *   the documentation and/or other materials provided with the         
+ *   distribution.                                                      
+ *                                                                      
+ * * Neither the name of the Brookhaven Science Associates, Brookhaven  
+ *   National Laboratory nor the names of its contributors may be used  
+ *   to endorse or promote products derived from this software without  
+ *   specific prior written permission.                                 
+ *                                                                      
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS  
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   
+ * POSSIBILITY OF SUCH DAMAGE.                                          
+ *
+ */
+
+#include <omp.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <stdint.h>
+
+#include "phocount.h"
+
+int count(data_t *in, data_t *out, data_t *stddev, 
+          int ndims, index_t *dims, 
+          data_t *thresh, data_t *mean_filter, int sum_max, int nan){
+  index_t nimages = dims[0];
```

Comment:
In gain correction code you use a more sophisticated method of finding nimages that (if I understood it right) let there be an arbitrary number of leading dimensions.